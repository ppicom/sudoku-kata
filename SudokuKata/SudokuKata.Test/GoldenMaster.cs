using ApprovalTests;
using ApprovalTests.Reporters;
using SudokuKata.Service;

namespace SudokuKata.Test;

[UseReporter(typeof(DiffReporter))]
public class GoldenMaster: IDisposable
{
    SudokuSolver solver;
    StringWriter writer;

    public GoldenMaster()
    {
        writer = new StringWriter();
        solver = new SudokuSolver();
        Console.SetOut(writer);
    }

    public void Dispose()
    {
        writer.Dispose();
        Console.Clear();
    }

    [Fact]
    public void TestReproductibility()
    {
        Console.Clear();
        Console.SetOut(writer);
        solver.Play();
        Approvals.Verify(writer);
    }
}
