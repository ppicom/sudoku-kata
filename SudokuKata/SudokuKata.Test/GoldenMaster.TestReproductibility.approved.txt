
Final look of the solved board:
+---+---+---+
|463|158|792|
|591|247|683|
|287|963|541|
+---+---+---+
|312|789|456|
|759|436|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+

Starting look of the board to solve:
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|...|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|..6|.7.|93.|
|...|.15|...|
|875|...|.6.|
+---+---+---+

================================================================================

(5, 3) can only contain 9.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|..6|.7.|93.|
|...|.15|...|
|875|...|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079006070930000015000875000060

(8, 3) can only contain 4.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|..6|.7.|93.|
|..4|.15|...|
|875|...|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079006070930004015000875000060

(7, 4) can only contain 8.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|..6|87.|93.|
|..4|.15|...|
|875|...|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079006870930004015000875000060

(7, 2) can only contain 2.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|.26|87.|93.|
|..4|.15|...|
|875|...|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079026870930004015000875000060

(7, 6) can only contain 4.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|.26|874|93.|
|..4|.15|...|
|875|...|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079026874930004015000875000060

(7, 1) can only contain 1.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|126|874|93.|
|..4|.15|...|
|875|...|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079126874930004015000875000060

(7, 9) can only contain 5.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|126|874|935|
|..4|.15|...|
|875|...|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079126874935004015000875000060

(9, 4) can only contain 3.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|126|874|935|
|..4|.15|...|
|875|3..|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079126874935004015000875300060

(8, 4) can only contain 6.
+---+---+---+
|4.3|15.|7..|
|...|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|3..|.6.|
+---+---+---+
Code: 403150700000207600007900040012080000009400010008020079126874935004615000875300060

(2, 3) can only contain 1.
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|3..|.6.|
+---+---+---+
Code: 403150700001207600007900040012080000009400010008020079126874935004615000875300060

(9, 5) can only contain 9.
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|.2.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|39.|.6.|
+---+---+---+
Code: 403150700001207600007900040012080000009400010008020079126874935004615000875390060

(6, 4) can only contain 5.
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|...|
|..9|4..|.1.|
|..8|52.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|39.|.6.|
+---+---+---+
Code: 403150700001207600007900040012080000009400010008520079126874935004615000875390060

(4, 8) can only contain 5.
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|.5.|
|..9|4..|.1.|
|..8|52.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|39.|.6.|
+---+---+---+
Code: 403150700001207600007900040012080050009400010008520079126874935004615000875390060

(9, 6) can only contain 2.
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|.8.|.5.|
|..9|4..|.1.|
|..8|52.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|.6.|
+---+---+---+
Code: 403150700001207600007900040012080050009400010008520079126874935004615000875392060

(4, 4) can only contain 7.
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|.4.|
+---+---+---+
|.12|78.|.5.|
|..9|4..|.1.|
|..8|52.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|.6.|
+---+---+---+
Code: 403150700001207600007900040012780050009400010008520079126874935004615000875392060

Column #7 can contain 5 only at (3, 7).
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|54.|
+---+---+---+
|.12|78.|.5.|
|..9|4..|.1.|
|..8|52.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|.6.|
+---+---+---+
Code: 403150700001207600007900540012780050009400010008520079126874935004615000875392060

Block (2, 2) can contain 9 only at (4, 6).
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|54.|
+---+---+---+
|.12|789|.5.|
|..9|4..|.1.|
|..8|52.|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|.6.|
+---+---+---+
Code: 403150700001207600007900540012789050009400010008520079126874935004615000875392060

Block (2, 2) can contain 1 only at (6, 6).
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|54.|
+---+---+---+
|.12|789|.5.|
|..9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|.6.|
+---+---+---+
Code: 403150700001207600007900540012789050009400010008521079126874935004615000875392060

Block (1, 3) can contain 1 only at (3, 9).
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|541|
+---+---+---+
|.12|789|.5.|
|..9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|.6.|
+---+---+---+
Code: 403150700001207600007900541012789050009400010008521079126874935004615000875392060

(9, 9) can only contain 4.
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|541|
+---+---+---+
|.12|789|.5.|
|..9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|.64|
+---+---+---+
Code: 403150700001207600007900541012789050009400010008521079126874935004615000875392064

(9, 7) can only contain 1.
+---+---+---+
|4.3|15.|7..|
|..1|2.7|6..|
|..7|9..|541|
+---+---+---+
|.12|789|.5.|
|..9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001207600007900541012789050009400010008521079126874935004615000875392164

Row #2 can contain 4 only at (2, 5).
+---+---+---+
|4.3|15.|7..|
|..1|247|6..|
|..7|9..|541|
+---+---+---+
|.12|789|.5.|
|..9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247600007900541012789050009400010008521079126874935004615000875392164

Column #1 can contain 7 only at (5, 1).
+---+---+---+
|4.3|15.|7..|
|..1|247|6..|
|..7|9..|541|
+---+---+---+
|.12|789|.5.|
|7.9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247600007900541012789050709400010008521079126874935004615000875392164

Row #2 can contain 3 only at (2, 9).
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|.12|789|.5.|
|7.9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541012789050709400010008521079126874935004615000875392164

(4, 9) can only contain 6.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|.12|789|.56|
|7.9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541012789056709400010008521079126874935004615000875392164

(4, 1) can only contain 3.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|.56|
|7.9|4..|.1.|
|..8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541312789056709400010008521079126874935004615000875392164

(6, 1) can only contain 6.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|.56|
|7.9|4..|.1.|
|6.8|521|.79|
+---+---+---+
|126|874|935|
|..4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541312789056709400010608521079126874935004615000875392164

(8, 1) can only contain 9.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|.56|
|7.9|4..|.1.|
|6.8|521|.79|
+---+---+---+
|126|874|935|
|9.4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541312789056709400010608521079126874935904615000875392164

(5, 2) can only contain 5.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|.56|
|759|4..|.1.|
|6.8|521|.79|
+---+---+---+
|126|874|935|
|9.4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541312789056759400010608521079126874935904615000875392164

(6, 2) can only contain 4.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|.56|
|759|4..|.1.|
|648|521|.79|
+---+---+---+
|126|874|935|
|9.4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541312789056759400010648521079126874935904615000875392164

(6, 7) can only contain 3.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|.56|
|759|4..|.1.|
|648|521|379|
+---+---+---+
|126|874|935|
|9.4|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541312789056759400010648521379126874935904615000875392164

(8, 2) can only contain 3.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|.56|
|759|4..|.1.|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541312789056759400010648521379126874935934615000875392164

(4, 7) can only contain 4.
+---+---+---+
|4.3|15.|7..|
|..1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|.1.|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|...|
|875|392|164|
+---+---+---+
Code: 403150700001247603007900541312789456759400010648521379126874935934615000875392164

(2, 1) can only contain 5.
+---+---+---+
|4.3|15.|7..|
|5.1|247|6.3|
|..7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|.1.|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|...|
|875|392|164|
+---+---+---+
Code: 403150700501247603007900541312789456759400010648521379126874935934615000875392164

(3, 1) can only contain 2.
+---+---+---+
|4.3|15.|7..|
|5.1|247|6.3|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|.1.|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|...|
|875|392|164|
+---+---+---+
Code: 403150700501247603207900541312789456759400010648521379126874935934615000875392164

Row #8 can contain 7 only at (8, 9).
+---+---+---+
|4.3|15.|7..|
|5.1|247|6.3|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|.1.|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|..7|
|875|392|164|
+---+---+---+
Code: 403150700501247603207900541312789456759400010648521379126874935934615007875392164

Guessing that 2 and 8 are arbitrary in column #9 (multiple solutions): Pick 2->(1, 9), 8->(5, 9).
+---+---+---+
|4.3|15.|7.2|
|5.1|247|6.3|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|.18|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|..7|
|875|392|164|
+---+---+---+
Code: 403150702501247603207900541312789456759400018648521379126874935934615007875392164

(5, 7) can only contain 2.
+---+---+---+
|4.3|15.|7.2|
|5.1|247|6.3|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|..7|
|875|392|164|
+---+---+---+
Code: 403150702501247603207900541312789456759400218648521379126874935934615007875392164

(8, 7) can only contain 8.
+---+---+---+
|4.3|15.|7.2|
|5.1|247|6.3|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|8.7|
|875|392|164|
+---+---+---+
Code: 403150702501247603207900541312789456759400218648521379126874935934615807875392164

(8, 8) can only contain 2.
+---+---+---+
|4.3|15.|7.2|
|5.1|247|6.3|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 403150702501247603207900541312789456759400218648521379126874935934615827875392164

Guessing that 8 and 9 are arbitrary in row #2 (multiple solutions): Pick 9->(2, 2), 8->(2, 8).
+---+---+---+
|4.3|15.|7.2|
|591|247|683|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 403150702591247683207900541312789456759400218648521379126874935934615827875392164

(1, 8) can only contain 9.
+---+---+---+
|4.3|15.|792|
|591|247|683|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|4..|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 403150792591247683207900541312789456759400218648521379126874935934615827875392164

Guessing that 3 and 6 are arbitrary in row #5 (multiple solutions): Pick 3->(5, 5), 6->(5, 6).
+---+---+---+
|4.3|15.|792|
|591|247|683|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|436|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 403150792591247683207900541312789456759436218648521379126874935934615827875392164

(1, 6) can only contain 8.
+---+---+---+
|4.3|158|792|
|591|247|683|
|2.7|9..|541|
+---+---+---+
|312|789|456|
|759|436|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 403158792591247683207900541312789456759436218648521379126874935934615827875392164

(3, 6) can only contain 3.
+---+---+---+
|4.3|158|792|
|591|247|683|
|2.7|9.3|541|
+---+---+---+
|312|789|456|
|759|436|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 403158792591247683207903541312789456759436218648521379126874935934615827875392164

(3, 5) can only contain 6.
+---+---+---+
|4.3|158|792|
|591|247|683|
|2.7|963|541|
+---+---+---+
|312|789|456|
|759|436|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 403158792591247683207963541312789456759436218648521379126874935934615827875392164

(1, 2) can only contain 6.
+---+---+---+
|463|158|792|
|591|247|683|
|2.7|963|541|
+---+---+---+
|312|789|456|
|759|436|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 463158792591247683207963541312789456759436218648521379126874935934615827875392164

(3, 2) can only contain 8.
+---+---+---+
|463|158|792|
|591|247|683|
|287|963|541|
+---+---+---+
|312|789|456|
|759|436|218|
|648|521|379|
+---+---+---+
|126|874|935|
|934|615|827|
|875|392|164|
+---+---+---+
Code: 463158792591247683287963541312789456759436218648521379126874935934615827875392164

